### PR TITLE
fix: type all buttons

### PR DESCRIPTION
**Pull request by estib-vega**
Apparently buttons without a type inside a form default to `submit`.
Example: Inside the Integrate Upstream modal, in order to open the links in the commit list and not trigger the upstream to be integrated, we need to set the type of the button to `button`

More info on the button.type: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#type

### CODE REVIEW

**Review by PavelLaptev — APPROVED** [review]

File: apps/desktop/src/lib/navigation/ProjectSelector.svelte [hunk 1]
```diff
@@ -6,22 +6,27 @@
...
+	}
+
+	let { isNavCollapsed }: Props = $props();
 
-	let buttonTrigger: HTMLButtonElement;
 	const project = getContext(Project);
 
-	let popup: ProjectsPopup;
+	let buttonTrigger = $state<HTMLButtonElement>();
+	let popup = $state<ReturnType<typeof ProjectsPopup>>();
```
Comment (estib-vega) [inline apps/desktop/src/lib/navigation/ProjectSelector.svelte:18]: Small drive-by fix
